File: ELA.CR.NewProposal/core/payloadtransferasset.go
package core

import "io"

type PayloadTransferAsset struct{}

func (a *PayloadTransferAsset) Data(version byte) []byte {
	//TODO: implement TransferAsset.Data()
	return []byte{0}
}

func (a *PayloadTransferAsset) Serialize(w io.Writer, version byte) error {
	return nil
}

func (a *PayloadTransferAsset) Deserialize(r io.Reader, version byte) error {
	return nil
}

File: SPV.CPP/SDK/ELACoreExt/Payload/PayloadTransferAsset.cpp
// Copyright (c) 2012-2018 The Elastos Open Source Project
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

#include "PayloadTransferAsset.h"

namespace Elastos {
	namespace SDK {

		PayloadTransferAsset::PayloadTransferAsset() {

		}

		PayloadTransferAsset::~PayloadTransferAsset() {

		}

		CMBlock PayloadTransferAsset::getData() const {
			//todo implement IPayload getData
			return CMBlock();
		}

		void PayloadTransferAsset::Serialize(ByteStream &ostream) const {

		}

		bool PayloadTransferAsset::Deserialize(ByteStream &istream) {
			return true;
		}

		nlohmann::json PayloadTransferAsset::toJson() {
			nlohmann::json jsonData;
			return jsonData;
		}

		void PayloadTransferAsset::fromJson(const nlohmann::json &jsonData) {

		}
	}
}